src/main/java/org/opensearch/security/securityconf/SecurityRoles.java

```diff
@@ -85,4 +85,6 @@ Set<String> getAllPermittedIndicesForDashboards(
 
     SecurityRoles filter(Set<String> roles);
 
+    Set<SecurityRole> getRoles();
```

Comment:
[Blocking] This breaks the encapsulation of this SecurityRoles, instead there should be a way to check for explicit permissions, something like, `boolean hasExplicitIndexPermissions(Resolved resolved, User user, String action, ...)`.

This would greatly reduce the amount of code in the SecurityIndexAccessEvaluator, and remove the other casting requirements in these ConfigModel classes.

Reply:
I have created a PR [1 ]that can address this, still needs cleanup but it can get you started
- https://github.com/samuelcostae/security/pull/1

It passes the following tests:
- SystemIndexDisabledTests
- SystemIndexPermissionDisabledTests	
- SystemIndexPermissionEnabledTests

Reply:
FYI - @DarshitChanpura @samuelcostae